backend/server/routers/courses.py

```diff
@@ -397,15 +399,10 @@ def get_path_from(course: str) -> dict[str, str | list[str]]:
     fetches courses which can be used to satisfy 'course'
     eg 2521 -> 1511
     """
-    course_condition = CONDITIONS.get(course)
-    if not course_condition:
-        raise HTTPException(400, f"no course by name {course}")
+    out: List[str] = list(INCOMING_ADJACENCY.get(course, []))
```

Comment:
This is not a list[str] yet. It might be None since that's the return type of .get().
That's probably why mypy is screaming at you. 
If the course is not found, you will get a list(None) which I think is a type error (?)

Reply:
You should add a HTTP exception in the case it is None.

Reply:
Actually now that I think about it, nvm this makes sense, I'm wrong.